xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -514,9 +462,9 @@ void CPeripheralBusAddon::PromptEnableAddons(const ADDON::VECADDONS& disabledAdd
   bool bAccepted = false;
 
   auto itAddon = std::find_if(disabledAddons.begin(), disabledAddons.end(),
-    [](const AddonPtr& addon)
+    [](const BinaryAddonBasePtr& addonInfo)
   {
-    return std::static_pointer_cast<CPeripheralAddon>(addon)->HasButtonMaps();
+    return addonInfo->Type(ADDON::ADDON_PERIPHERALDLL)->GetValue("@provides_buttonmaps").asBoolean();
```

Comment:
Makes sense, maybe a static function HasButtonMaps() that takes the xml parameter?